src/ngraph/runtime/dynamic_wrapper/dynamic_wrapper_backend.cpp

```diff
@@ -0,0 +1,277 @@
+        const element::Type& type, const PartialShape& shape)
+{
+    return make_shared<WrappedDynamicTensor>(type, shape, m_wrapped_backend);
+}
+
+shared_ptr<runtime::Executable>
+    runtime::dynamic_wrapper::DynamicWrapperBackend::compile(shared_ptr<Function> function,
+                                                             bool enable_performance_collection)
+{
+    return make_shared<runtime::dynamic_wrapper::WrappedExecutable>(
+        function, m_wrapped_backend, enable_performance_collection);
+}
+
+runtime::dynamic_wrapper::WrappedExecutable::WrappedExecutable(
+    shared_ptr<Function> wrapped_function,
+    shared_ptr<runtime::Backend> wrapped_backend,
+    bool enable_performance_collection)
+    : m_wrapped_function(wrapped_function)
+    , m_wrapped_backend(wrapped_backend)
+    , m_enable_performance_collection(enable_performance_collection)
+{
+    pass::Manager passes;
+    passes.register_pass<pass::ShapeRelevance>();
+    passes.run_passes(m_wrapped_function);
+
+    set_parameters_and_results(*wrapped_function);
+}
+
+bool runtime::dynamic_wrapper::WrappedExecutable::call(
+    const std::vector<std::shared_ptr<runtime::Tensor>>& outputs,
+    const std::vector<std::shared_ptr<runtime::Tensor>>& inputs)
+{
+    // TODO: Get cached executable out if it exists.
+    // We will cache on:
+    // (1) all shapes;
+    // (2) all values of shape-relevant input tensors.
+
+    std::vector<std::shared_ptr<runtime::Tensor>> wrapped_inputs;
+    std::vector<element::Type> arg_element_types;
+    std::vector<PartialShape> arg_shapes;
+
+    for (auto& input : inputs)
+    {
+        if (auto static_tensor =
+                std::dynamic_pointer_cast<runtime::dynamic_wrapper::WrappedStaticTensor>(input))
+        {
+            arg_element_types.push_back(static_tensor->get_wrapped_tensor()->get_element_type());
+            arg_shapes.push_back(static_tensor->get_wrapped_tensor()->get_shape());
+            wrapped_inputs.push_back(static_tensor->get_wrapped_tensor());
+        }
+        else if (auto dynamic_tensor =
+                     std::dynamic_pointer_cast<runtime::dynamic_wrapper::WrappedDynamicTensor>(
+                         input))
+        {
+            NGRAPH_CHECK(dynamic_tensor->has_storage());
+            arg_element_types.push_back(dynamic_tensor->get_wrapped_tensor()->get_element_type());
+            arg_shapes.push_back(dynamic_tensor->get_wrapped_tensor()->get_shape());
+            wrapped_inputs.push_back(dynamic_tensor->get_wrapped_tensor());
+        }
+        else
+        {
+            NGRAPH_CHECK(false,
+                         "Internal error: Tensor is neither a WrappedStaticTensor nor a "
+                         "WrappedDynamicTensor");
+        }
+    }
+
+    // TODO: specialize_shapes needs to fill in values of shape-relevant params.
+    auto clone = specialize_shapes(m_wrapped_function, arg_element_types, arg_shapes);
```

Comment:
Can this be invoked through pass manager ?

Reply:
Should be doable. Let me poke on it and see how big a change it will be---I might want to save it for a later PR.

Reply:
@nmostafa I think I will actually punt on this for now, since it will need to wait until after #2875 goes in.